### DISCUSSION TITLE
Format long numbers in table and keep them sortable

### DISCUSSION THREAD

**Discussion by Settleman** [EvotecIT/PSWriteHTML#373] [Q&A]
Hi,

I want to show file size details in a report but I want to show them in MB and keep them sortable.
I changed the contents to a string but the sorting is text based and won't work as expected so is there a way to show the formatted text from a long number?

For instance, using this dataset:
Get-Process | select ProcessName, PeakVirtualMemorySize -first 10

From this number representing bytes 1454019 can I show it as "1,419.94 MB" or just "1,419.94" and keep it sortable?

If this is not possible, can you suggest an approach to achieve this goal?

Thanks in advance.

C

![image](https://user-images.githubusercontent.com/68226552/222422512-7bc42589-c066-414c-983c-0bc3b4a87a83.png)

**Comment by PrzemyslawKlys** [comment]
There's no way to do it natively now. You could use Mailbox Size (MB) as a column and sort the rest as a Number without MB.

DataTables have https://datatables.net/plug-ins/sorting/ plugins which can be added but would require some work on implementation. Plugins ideas:
- https://datatables.net/plug-ins/sorting/file-size
- https://datatables.net/plug-ins/sorting/formatted-numbers
- https://datatables.net/plug-ins/sorting/numeric-comma

But I don't have time at the moment to take on those. So unless someone steps in - you're going to have to workaround it.

**Reply by Settleman** [reply]
I totally understand mate. I'm juggling myself, we're 3 engineers down on a team of 5 and having to do interviews! It's mental! :/
I will just use numbers for now, thanks.

C
